src/tpm2.c

```diff
@@ -5648,6 +5675,67 @@ UINT16 TPM2_GetVendorID(void)
     return vid;
 }
 
+/* Stores nameAlg + the digest of nvPublic in buffer, total size in size */
+int TPM2_HashNvPublic(TPMS_NV_PUBLIC* nvPublic, byte* buffer, UINT16* size)
+{
+    int rc;
+#ifndef WOLFTPM2_NO_WOLFCRYPT
+    int hashSize, nameAlgValue, nameAlgSize;
+    wc_HashAlg hash;
+    enum wc_HashType hashType;
+    byte appending[sizeof(TPMS_NV_PUBLIC)];
+    TPM2B_DIGEST digest;
+    TPM2_Packet packet;
+
+    /* Prepare temporary buffer */
+    packet.buf = appending;
+    packet.pos = 0;
+    packet.size = sizeof(appending);
+
+    /* nvPublic must be in Marshaled state for hashing */
+    TPM2_Packet_AppendU32(&packet, nvPublic->nvIndex);
+    TPM2_Packet_AppendU16(&packet, nvPublic->nameAlg);
+    TPM2_Packet_AppendU32(&packet, nvPublic->attributes);
+    TPM2_Packet_AppendU16(&packet, nvPublic->authPolicy.size);
+    TPM2_Packet_AppendBytes(&packet, nvPublic->authPolicy.buffer,
+        nvPublic->authPolicy.size);
+    TPM2_Packet_AppendU16(&packet, nvPublic->dataSize);
+
+    /* Hashing nvPublic */
+    rc = TPM2_GetHashType(nvPublic->nameAlg);
+    hashType = (enum wc_HashType)rc;
+    rc = wc_HashGetDigestSize(hashType);
+    if (rc < 0) {
+        return rc;
+    }
+    hashSize = rc;
+
+    rc = wc_HashInit(&hash, hashType);
+    if(rc == 0) {
+        rc = wc_HashUpdate(&hash, hashType, packet.buf, packet.pos);
+    }
+
+    if (rc == 0) {
+        rc = wc_HashFinal(&hash, hashType, digest.buffer);
+    }
+
+    if (rc == 0) {
+        nameAlgValue = TPM2_Packet_SwapU16(nvPublic->nameAlg);
+        nameAlgSize = sizeof(nvPublic->nameAlg);
+        XMEMCPY(buffer, (byte*)&nameAlgValue, nameAlgSize);
+        XMEMCPY(&buffer[2], digest.buffer, hashSize);
+        /* account for nameAlg concatenation */
+        *size = hashSize + nameAlgSize;
+    }
+
+    wc_HashFree(&hash, hashType);
+
+    return TPM_RC_SUCCESS;
```

Comment:
Return code is not being checked or returned... Where is `rc` checked?

Reply:
this is changed to return rc; now

and in the case when wolfcrypt is not enabled(param encryption is not enabled) I return TPM_RC_SUCCESS, because the computation will not be used without param encryption.

Reply:
the RC result is returned from the wolfTPM2_wrappers to the caller

this is why the RC is not checked, I will add comment so it is clear